### PR TITLE
Create chats, send messages, asynchronous frontend

### PR DESCRIPTION
**Pull request by Hernqvist**
### Issues:
#89 #86 #98 #38

### Test Plan
Have tested with latest backend from master.

### Description
Can now create chats and send messages to chats. Frontend is now asynchronous. There is also an incomplete login screen that always allows you to log in, since login is not yet supported in backend. All messages are also sent by user id 1, will change this once we are clear on how login works.

When messages are sent or chats are created all data is reloaded from backend, which is a temporary solution before getting new messages by web sockets.

### CODE REVIEW

File: client/js/postManager.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        
+    constructor(myUrl) {
+        this.url = myUrl;
+    }
+    
+    postData(inUrl, inData, callback) {
+        $.ajax({
+            url: this.url+"/"+inUrl,
+            type: "POST",
+            data: JSON.stringify(inData),
```
Comment (Hernqvist) [inline client/js/postManager.js]: stringify doesn't seem to be redundant, if removed it doesn't work

**Comment by simonlindholm** [comment]
Can you squash the commits?

**Comment by s4wny** [comment]
I squashed the commits.

**Comment by Hernqvist** [comment]
Uh, why does it now say that Victor did everything? @s4wny

**Comment by s4wny** [comment]
@Hernqvist arrgh >.> The squash command changed that....

Maybe you can change the history with rebase so the commits was written by you?

@simonlindholm Does it exist any good solution that squash commits and still somehow keep track of which commits each person wrote?

**Comment by simonlindholm** [comment]
rebase ought to preserve ownership on the commit you squash everything into, but of course, that of the other commits gets lost. You can change the owner of the latest to something else using `git commit --amend --author="..."` (within an interactive rebase, if you want to change older commits). On the other hand, authorship is typically not that important on large collaborative branches that get squashed, and you could also just note the full list of authors in the commit message and/or issue.

**Pull request merged**